addons/mrp/models/mrp_production.py

```diff
@@ -803,6 +803,20 @@ def unlink(self):
             workorders_to_delete.unlink()
         return super(MrpProduction, self).unlink()
 
+    def copy(self, default=None):
+        new_mo = super(MrpProduction, self).copy(default=default)
+        # cover 2 cases: backorders generation (follow default logic for moves copying)
+        # and copying a done MO via the form (i.e. copy only the non-cancelled moves since no backorder = cancelled finished moves)
+        if not default or default.get('move_finished_ids', False) is not None:
+            finished_moves = self.move_finished_ids.filtered(lambda m: m.state != 'cancel' and m.product_qty != 0.0)
+            for move in finished_moves:
+                move.copy(default={'production_id': new_mo.id})
```

Comment:
Instead of copy, use copy_data and create moves in batch at the end. :rocket:

Reply:
Done :boom: